client/my-sites/stats/stats-upsell-modal/index.tsx

```diff
@@ -23,20 +23,18 @@ import './style.scss';
 export default function StatsUpsellModal( { siteId }: { siteId: number } ) {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const plan = useSelector( ( state ) => getPlanBySlug( state, PLAN_PREMIUM ) );
 	const selectedSiteId = useSelector( getSelectedSiteId );
-	const planMonthly = Plans.usePricingMetaForGridPlans( {
+	const siteSlug = useSelector( getSelectedSiteSlug );
+	const plans = Plans.usePlans( { coupon: undefined } );
+	const plan = plans?.data?.[ PLAN_PREMIUM ];
```

Comment:
It would, but I like the spirit/intention of moving more and more things/data closer to a single source (data-stores).

Follow up from here could be to be stripping out things from calypso-products that are served from data-stores. I'm not sure either if we have a case where we need any plan data be statically defined in calypso (in which case, we can also define localised selectors in the data-store)